src/compas/datastructures/mesh/subdivision.py

```diff
@@ -131,7 +149,7 @@ def mesh_subdivide_tri(mesh, k=1):
 
     """
     for _ in range(k):
-        subd = mesh.copy()
+        subd = mesh_fast_copy(mesh)
```

Comment:
have you tested this? because one of the reasons i had not yet applied this everywhere was because it sometimes failed....